src/anet.c

```diff
@@ -491,15 +492,34 @@ int anetUnixServer(char *err, char *path, mode_t perm, int backlog)
     return s;
 }
 
+/* Accept a connection and also make sure the socket is non-blocking, and CLOEXEC.
+ * returns the new socket FD, or -1 on error. */
 static int anetGenericAccept(char *err, int s, struct sockaddr *sa, socklen_t *len) {
     int fd;
     do {
+        /* Use the accept4() call on linux to simultaneously accept and
+         * set a socket as non-blocking. */
+#ifdef HAVE_ACCEPT4
+        fd = accept4(s, sa, len,  SOCK_NONBLOCK | SOCK_CLOEXEC);
+#else
         fd = accept(s,sa,len);
+#endif
     } while(fd == -1 && errno == EINTR);
     if (fd == -1) {
         anetSetError(err, "accept: %s", strerror(errno));
         return ANET_ERR;
     }
+#ifndef HAVE_ACCEPT4
+    if (anetCloexec(fd) == -1) {
+        anetSetError(err, "anetCloexec: %s", strerror(errno));
+        close(fd);
+        fd = -1;
```

Comment:
I suppose ANET_ERR = -1, but all of the callers of these functions check for ANET_ERR instead of -1 and we return ANET_ERR above. It would probably more maintainable to return ANET_ERR here and update the function comment.

Reply:
i think it is very ugly that we use ANET_ERR here, this function won't return ANET_OK, it returns an FD, and an invalid FD is -1.
but i see your point about consistency, i'll comply.